src/sentry_plugins/trello/plugin.py

```diff
@@ -0,0 +1,250 @@
+                "required": False,
+            },
+        ]
+
+    def get_client(self, project):
+        return TrelloApiClient(
+            self.get_option("key", project), token=self.get_option("token", project)
+        )
+
+    def error_message_from_json(self, data):
+        errors = data.get("errors")
+        if errors:
+            return " ".join([e["message"] for e in errors])
+        return "unknown error"
+
+    def create_issue(self, request, group, form_data, **kwargs):
+        client = self.get_client(group.project)
+
+        try:
+            response = client.new_card(
+                idList=form_data["list"], name=form_data["title"], desc=form_data["description"]
+            )
+        except Exception as e:
+            self.raise_error(e)
+
+        return response["shortLink"]
+
+    def link_issue(self, request, group, form_data, **kwargs):
+        client = self.get_client(group.project)
+
+        try:
+            card = client.get_card(form_data["issue_id"])
+        except Exception as e:
+            self.raise_error(e)
+
+        comment = form_data.get("comment")
+        if comment:
+            try:
+                client.create_comment(card["shortLink"], comment)
+            except Exception as e:
+                self.raise_error(e)
+
+        return {"title": card["name"], "id": card["shortLink"]}
+
+    def get_issue_label(self, group, issue, **kwargs):
+        # the old version of the plugin stores the url in the issue
+        if re.search("\w+/https://trello.com/", issue):
+            short_issue = issue.partition("/")[0]
+            return "Trello-%s" % short_issue
+        return "Trello-%s" % issue
+
+    def get_issue_url(self, group, issue, **kwargs):
+        # TODO(Steve): figure out why we sometimes get a string and sometimes a dict
+        if isinstance(issue, dict):
+            issue = issue["id"]
+        # the old version of the plugin stores the url in the issue
+        if re.search("\w+/https://trello.com/", issue):
+            url = issue.partition("/")[2]
+            return url
+        return "https://trello.com/c/%s" % issue
+
+    def view_options(self, request, group, **kwargs):
+        field = request.GET.get("option_field")
+        board = request.GET.get("board")
+
+        results = []
+        if field == "list" and board:
+            client = self.get_client(group.project)
+            try:
+                response = client.get_lists_of_board(board)
+            except Exception as e:
+                return Response(
+                    {
+                        "error_type": "validation",
+                        "errors": [{"__all__": self.message_from_error(e)}],
+                    },
+                    status=400,
+                )
+            else:
+                results = self.map_to_options(response)
+
+        return Response({field: results})
+
+    def view_autocomplete(self, request, group, **kwargs):
```

Comment:
How do you feel about adding docstrings to new methods? Will make it much easier to read, imo. Feel like it would be great to have docstrings on `IssueTrackingPlugin2`'s methods too but that can wait. [This](https://github.com/getsentry/sentry/blob/master/src/sentry/tsdb/base.py), I feel, is a pretty good example of good docstrings.

Reply:
@manuzope I'll add docstrings to the code I've added, good suggestion